.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,237 @@
+# Pre-requisites:
+# - AWS Account with correct permissions
+# - `gh-pages` branch
+# - AWS Account needs to add the LOG_GROUP_NAME defined below
+
+name: Soak Testing
+on:
+  workflow_dispatch:
+    inputs:
+      target_commit_sha:
+        description: 'The commit SHA on this repo to use for the Soak Tests.'
+        required: true
+      test_duration_minutes:
+        description: 'The duration of the Soak Tests in minutes.'
+        required: true
+        default: 300
+  schedule:
+    - cron: '0 15 * * *'
```

Comment:
Is it necessary to run this daily? Could this just be triggered to run after specific commits?

Reply:
We've seen the memory/cpu be different on the same commit but on a different day. It's probably because of GitHub Runners, but this gives us confidence in the results.

I actually think more scheduled workflows is a good idea. Some main workflows on our repos succeed initially but then 2 months later we run them and they fail. By running it on a schedule, the Soak Tests build the app every time from scratch by pulling in the latest from upstream OTel Ruby. This means we can detect failures much more early on and can be more sensitive to their failures when it may take weeks or months to add a new commit which would run these tests again 🙂